### DISCUSSION TITLE
Possible unhandled promise rejection

### DISCUSSION THREAD

**Discussion by proambb** [hyochan/react-native-iap#2891] [Q&A]
Thanks for the great library!

Every time I open the app in the Android emulator, I see this warning: "Possible unhandled promise rejection..." (see more details in the screenshot). I understand that this library is not expected to work on the Android emulator without installed Google Play services.

The issue isn't that the library isn't working, but rather the warning itself. It's not possible to catch and conditionally ignore it.

The error is thrown in [withIAPContext.ts](https://github.com/hyochan/react-native-iap/blob/615368b1fdee4eb082ea0b5a59e3ee1c0d59d3f9/src/hooks/withIAPContext.tsx#L133) file. There is no way to pass on error handlers to this HOC. 

`purchaseUpdatedListener` expects the second argument to be a function, if a function is not given it throws an error which causes the mentioned warning.

Of course, we can conditionally wrap with this HOC, for example detect Android emulator and don't wrap at all, but in this case the library won't work on emulators with installed GooglePlay services. 

Another possible solution is to allow passing on error handler to `withIAPContext` HOC. 

Are there any other ways to ignore this error? 

![image](https://github.com/user-attachments/assets/3bd1df61-f645-4411-9dac-15f498965c2c)

**Comment by hyochan** [comment]
`withIAPContext` HOC has been **removed** in react-native-iap v14+.

You are using an older version (v12 or v13). Please upgrade to v14+ which uses the new `useIAP` hook instead.

## Migration

In v14+, use the `useIAP` hook directly with error handling:

```typescript
import { useIAP } from 'react-native-iap';

const MyComponent = () => {
  const { connected, products } = useIAP({
    onPurchaseError: (error) => {
      // Handle errors here - no unhandled promise rejections
      console.log('Purchase error:', error);
    },
  });

  // Your component logic
};
```

## Benefits of v14+

- No more `withIAPContext` HOC
- Built-in error handling via `onPurchaseError` callback
- Cleaner API with `useIAP` hook
- Better emulator handling

## Migration Guide

Please follow the migration guide to upgrade:

- [Migration Guide (v13 to v14)](https://hyochan.github.io/react-native-iap/docs/guides/migration-v13-to-v14)

## Note

Versions below v14 are **no longer supported**. Use at your own risk.
